File: frontend/src/services/bookingService.ts
import apiClient from './api';
import { ApiResponse, PageResponse, Booking, BookingCreateRequest, PriceCalculationRequest, PriceBreakdownResponse } from '../types';

export const bookingService = {
  async calculatePrice(data: PriceCalculationRequest): Promise<PriceBreakdownResponse> {
    const response = await apiClient.post<ApiResponse<PriceBreakdownResponse>>('/bookings/calculate-price', data);
    return response.data.data;
  },

  async createBooking(data: BookingCreateRequest): Promise<Booking> {
    const response = await apiClient.post<ApiResponse<Booking>>('/bookings', data);
    return response.data.data;
  },

  async getMyBookings(page = 0, size = 10): Promise<PageResponse<Booking>> {
    const response = await apiClient.get<ApiResponse<PageResponse<Booking>>>('/bookings', { params: { page, size } });
    return response.data.data;
  },

  async getBookingById(id: number): Promise<Booking> {
    const response = await apiClient.get<ApiResponse<Booking>>(`/bookings/${id}`);
    return response.data.data;
  },

  async cancelBooking(id: number, reason?: string): Promise<Booking> {
    const response = await apiClient.put<ApiResponse<Booking>>(`/bookings/${id}/cancel`, null, { params: { reason } });
    return response.data.data;
  },
};

export default bookingService;

File: backend/src/main/java/com/driveease/service/BookingService.java
package com.driveease.service;

import com.driveease.dto.request.BookingCreateRequest;
import com.driveease.dto.request.PriceCalculationRequest;
import com.driveease.dto.response.BookingResponse;
import com.driveease.dto.response.PageResponse;
import com.driveease.dto.response.PriceBreakdownResponse;
import com.driveease.entity.enums.BookingStatus;

public interface BookingService {
    PriceBreakdownResponse calculatePrice(PriceCalculationRequest request);
    BookingResponse createBooking(Long userId, BookingCreateRequest request);
    PageResponse<BookingResponse> getUserBookings(Long userId, int page, int size, BookingStatus status);
    BookingResponse getBookingById(Long userId, Long bookingId);
    BookingResponse cancelBooking(Long userId, Long bookingId, String reason);
}
